pipeline.wdl

```diff
@@ -0,0 +1,246 @@
+            input:
+                sample = sample,
+                outputDirectory = outputDirectory + "/" + sample.id,
+                genomeFile = referenceGenome,
+                spliceJunctionsFile = select_first([spliceJunctionsFile, createSJsfile.outputSJsFile]),
+                dockerImages = dockerImages
+        }
+    }
+
+    call RunTalonOnLoop as runTalon {
+        input:
+            SAMfiles = flatten(sampleWorkflow.outputTranscriptCleanSAM),
+            configFile = talonConfigFile,
+            databaseFile = select_first([talonDatabase, createDatabase.outputDatabase]),
+            outputPrefix = outputDirectory + "/" + pipelineRunName,
+            genomeBuild = genomeBuild,
+            sequencingPlatform = sequencingPlatform,
+            dockerImage = dockerImages["talon"]
+    }
+
+    call talon.CreateAbundanceFileFromDatabase as createAbundanceFile {
+        input:
+            databaseFile = runTalon.outputUpdatedDatabase,
+            outputPrefix = outputDirectory + "/" + pipelineRunName,
+            genomeBuild = genomeBuild,
+            annotationVersion = annotationVersion,
+            filterTranscripts = filterTranscriptsAbundance,
+            filterPairingsFile = filterPairingsFileAbundance,
+            dockerImage = dockerImages["talon"]
+    }
+
+    call talon.SummarizeDatasets as createSummaryFile {
+        input:
+            databaseFile = runTalon.outputUpdatedDatabase,
+            outputPrefix = outputDirectory + "/" + pipelineRunName,
+            datasetGroupsCSV = summaryDatasetGroupsCSV,
+            dockerImage = dockerImages["talon"]
+    }
+
+    output {
+        File outputSpliceJunctionsFile = select_first([spliceJunctionsFile, createSJsfile.outputSJsFile])
+        Array[File] outputMinimap2 = flatten(sampleWorkflow.outputMinimap2)
+        Array[File] outputTranscriptCleanFasta = flatten(sampleWorkflow.outputTranscriptCleanFasta)
+        Array[File] outputTranscriptCleanLog = flatten(sampleWorkflow.outputTranscriptCleanLog)
+        Array[File] outputTranscriptCleanSAM = flatten(sampleWorkflow.outputTranscriptCleanSAM)
+        Array[File] outputTranscriptCleanTElog = flatten(sampleWorkflow.outputTranscriptCleanTElog)
+        File outputTalonDatabase = runTalon.outputUpdatedDatabase
+        File outputTalonLog = runTalon.outputLog
+        File outputAbundance = createAbundanceFile.outputAbundanceFile
+        File outputSummary = createSummaryFile.outputSummaryFile
+    }
+}
+
+task RunTalonOnLoop {
+    input {
+        Array[File] SAMfiles
+        File configFile
+        File databaseFile
+        String outputPrefix
+        String genomeBuild
+        String sequencingPlatform = "PacBio-Sequal"
+
+        Float? minimumCoverage = 0.9
+        Int? minimumIdentity = 0
+
+        Int cores = 1
+        Int memory = 20
+        String dockerImage = "biocontainers/talon:v4.2_cv2"
+    }
+
+    command <<<
+        #$ -l h_vmem=~{memory}G
+        #$ -pe BWA ~{cores}
```

Comment:
Why are these two lines here? This type of stuff should be handled by the runtime attributes.

Reply:
Part of a fix that was not actually needed, removed them now.